### PR TITLE
pkg/plugins: rename to pkg/kustomize

### PR DESCRIPTION
**Pull request by ykaliuta**
Change naming to make it more suitable for future kustomization

## Description
Plugin here is a term in kustomize domain. For the operator it amends the manifests and particular kustomize tools used for that (plugins, transformers) is an extra implementation detail.

This will allow to decouple functionality in operator domain from kustomize (add replicas, images but not PatchTransformer or ReplacementTransformer) and allow implement own amendments (or custom kustomize transformers) using low level kustomize tools if needed.

Rename functions to represent intentions as well:

ApplyNamespacePlugin -> TransformNamespace
ApplyAddLabelsPlugin -> AddLabels

No functionality changes.

## How Has This Been Tested?

No functionality changes. Github testing is fine.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/patches/patches.go [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+				Path:               "metadata/labels",
+				CreateIfNotPresent: true,
+			},
+		},
+	}
+
+	return nsplug.Transform(resMap)
+}
+
+func AddNamespace(manifestNamespace string, resMap resmap.ResMap) error {
```
[the remark this answers is not in the document]
Comment (zdtsw) [reply pkg/patches/patches.go:39]: 2 minor comments for the name of the functions.

**Review by zdtsw — APPROVED** [review]
/lgtm

**Review by ajaypratap003 — COMMENTED** [review]
LGTM

**Comment by bartoszmajsak** [comment]
I am not quite sure I am convinced by this change. Plugin adds functionality whereas a patch is more of an implementation detail of how this functionality is done.

In addition, maybe we could consider a bit different renaming revealing the intent and not what's happening under the hood:
  * `ApplyAddLabelsPlugin` -> `AddLabels` 
  * `ApplyNamespacePlugin` -> `TransformNamespace`

I know naming is hard, but I am not sure with this rename we are making it any easier for a newcomer to understand what is the responsibility of this package.

**Comment by VaishnaviHire** [comment]
> I am not quite sure this change convinces me. Plugin adds functionality whereas a patch is more of an implementation detail of how this functionality can be brought to life.
> 
> In addition, maybe we could consider a bit different renaming that reveals the intent and not what's happening under the hood:
> 
> * `ApplyAddLabelsPlugin` -> `AddLabels`
> * `ApplyNamespacePlugin` -> `TransformNamespace`
> 
> I know naming is hard, but I am not sure with this rename we are making it any easier for a newcomer to understand what is the responsibility of this package. Perhaps putting it under `pkg/kustomize/plugins` would make the structure more accurate.
> 
> Especially that from what I understand, the replicas used as an example in the PR description, will also come as a kustomize plugin:
> 
> > The idea is to create a custom plugin for the kustomize where we use JSON patch to patch these updated values and then deploy the manifests.
> 
> Just my $.2

+1 `plugins` represent here the `kustomize go plugins` that will be used. Here is the documentation of the plugins that are being used and can be used in future - https://github.com/kubernetes-sigs/kustomize/tree/master/plugin/builtin

**Comment by ykaliuta** [comment]
> 
> +1 `plugins` represent here the `kustomize go plugins` that will be used. Here is the documentation of the plugins that are being used and can be used in future - https://github.com/kubernetes-sigs/kustomize/tree/master/plugin/builtin

Which it shouldn't IMHO. Imagine I want to use patchjson6902transformer for, for example, replica patching. Depends on some condition. And then I want to patch some other field and using the same transformer. Should it be one call to the `apply` function? Imagine I changed my implementation of the replica patching to use another plugin. Should I then rename file and the function?

**Comment by LaVLaS** [comment]
I am a +1 to keeping `plugins` since it seems like the current and future intent is to continue using the `kustomize/api/builtins` for these functions.  If there were a change to the underlying implementation away from kustomize plugins, I think it would be better to create a new directory for that functionality that clearly demonstrates the change in implementation.

> In addition, maybe we could consider a bit different renaming that reveals the intent and not what's happening under the hood:
>
> * ApplyAddLabelsPlugin -> AddLabels
> * ApplyNamespacePlugin -> TransformNamespace

**[If we have to change the function names]** As someone who just started a deep dive into this code base +1 to this recommendation for the function names as it clearly shows the intent without referencing the internal implementation.  It makes it clear that you are performing an additive action on labels and transforming the namespaces across the target resources.

**Comment by ykaliuta** [comment]
> I am a +1 to keeping `plugins` since it seems like the current and future intent is to continue using the `kustomize/api/builtins` for these functions. If there were a change to the underlying implementation away from kustomize plugins, I think it would be better to create a new directory for that functionality that clearly demonstrates the change in implementation.

Why caller should care about implementation (it cares due to module name)?

> 
> > In addition, maybe we could consider a bit different renaming that reveals the intent and not what's happening under the hood:
> > 
> > * ApplyAddLabelsPlugin -> AddLabels
> > * ApplyNamespacePlugin -> TransformNamespace
> 
> **[If we have to change the function names]** As someone who just started a deep dive into this code base +1 to this recommendation for the function names as it clearly shows the intent without referencing the internal implementation. It makes it clear that you are performing an additive action on labels and transforming the namespaces across the target resources.

Ok. A bit better.
Now again. See my experiments with environment patching https://paste.debian.net/1305963. Imagine we should go this way. Where to put it?

Imagine the implementation changed to some plugin.

**Comment by ykaliuta** [comment]
I would really understand `plugins` module if we put there own extension of kustomize with own plugins.

**Comment by ykaliuta** [comment]
Will that work?

**Comment by ykaliuta** [comment]
@bartoszmajsak @LaVLaS  will the change acceptable to you? I kept `kustomize`  and renamed the functions like you suggested.

**Comment by bartoszmajsak** [comment]
> @bartoszmajsak @LaVLaS  will the change be acceptable to you? I kept `kustomize`  and renamed the functions like you suggested.

@ykaliuta I'm on PTO this week. Back on Monday.

**Pull request closed without merging**